app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt

```diff
@@ -265,6 +244,26 @@ class ThreadFragment : Fragment() {
         phoneContextualMenuAlertDialog.initValues(snackBarManager)
     }
 
+    private fun observeThreadOpening() = with(threadViewModel) {
+
+        mainViewModel.currentThreadUid.distinctUntilChanged().observeNotNull(viewLifecycleOwner) { threadUid ->
+
+            reassignThreadLive(threadUid)
+            reassignMessagesLive(threadUid)
+
+            openThread(threadUid).observe(viewLifecycleOwner) { result ->
```

Comment:
This `openThread()` method needs the updated Thread object to work. Shouldn't we wait for `reassignThreadLive` to have finished before calling `openThread()` so we can skip on sending some of the `threadUid` values that are passed around everywhere and use the `threadLive` latest value instead?

Reply:
How do you want to do that exactly? I don't see any elegant way.